controllers/provisioners/eks/update.go

```diff
@@ -164,11 +164,15 @@ func (ctx *EksInstanceGroupContext) UpdateScalingGroup(configName string, scalin
 				Version:            aws.String("$Latest"),
 			}
 		}
-		latestVersion := (*scalingConfig).(*scaling.LaunchTemplate).LatestVersion.VersionNumber
-		latestVersionString := strconv.FormatInt(*latestVersion, 10)
 
 		status.SetActiveLaunchTemplateName(configName)
-		status.SetLatestTemplateVersion(latestVersionString)
+		switch scalingConfigType := (*scalingConfig).(type) {
```

Comment:
So in this case spec.IsLaunchTemplate() is true because spec has changed to LaunchTemplate, but scalingConfig is not yet created? what does that mean for the rolling upgrade?

Reply:
It's more that the scalingConfig exists, but it's of type *scaling.LaunchConfig.

Just tested the rolling upgrade out - there's one small issue in that two CRs get submitted (with the default concurrency policy). One like `rollup-vault-20210208201848-0`, and another like `rollup-vault-20210208201848-1`. This is probably due to our defaulting the LaunchTemplate version to '0' if it's not set.

Reply:
We could probably just default it to 1 in the case that scalingConfig is of type *scaling.LaunchConfig

Reply:
What happens when we go from LaunchTemplate back to LaunchConfig?
We should also check the case of MixedPolicy

Reply:
Great question - I haven't tried that case :) - though I don't think this PR would change it, since this is only running if type==LaunchTemplate. Let me check real quick though!

Reply:
LC -> LT works fine